lib/models/blueprint.js

```diff
@@ -1046,29 +1049,57 @@ Blueprint.prototype.addBowerPackagesToProject = function(packages, installOption
   @return {Promise}
 */
 Blueprint.prototype.addAddonToProject = function(options) {
-  var taskOptions = {};
+  return this.addAddonsToProject({
+    'packages': [options],
+    extraArgs: options.extraArgs || {},
+    blueprintOptions: options.blueprintOptions || {}
+  });
+};
 
-  if (typeof options === 'string') {
-    taskOptions['packages'] = [options];
-  } else {
-    if (!options.name) {
-      throw new SilentError('You must provide a `name` to addAddonToProject');
-    }
+/**
+  Used to add multiple addons to the project's `package.json` and run their
+  `defaultBlueprint` if they provide one.
 
-    if (options.target) {
-      options.name += '@' + options.target;
-    }
+  Generally, this would be done from the `afterInstall` hook, to
+  ensure that a package that is required by a given blueprint is
+  available.
 
-    taskOptions['packages']      = [options.name];
-    taskOptions.extraArgs        = options.extraArgs || [];
-    taskOptions.blueprintOptions = options.blueprintOptions || {};
-  }
+  @method addAddonsToProject
+  @param {Object} options
+  @return {Promise}
+*/
+Blueprint.prototype.addAddonsToProject = function (options) {
+  var taskOptions = {
+    'packages': [],
+    extraArgs: options.extraArgs || [],
+    blueprintOptions: options.blueprintOptions || {}
+  };
 
-  var task = this.taskFor('addon-install');
+  var packages = options['packages'];
+  if (packages && packages.length) {
+    taskOptions['packages'] = packages.map(function (pkg) {
+      if (typeof pkg === 'string') {
+        return pkg;
+      }
 
-  this._writeStatusToUI(chalk.green, 'install addon', taskOptions['packages']);
+      if (!pkg.name) {
+        throw new SilentError('You must provide a package `name` to addAddonsToProject');
+      }
+
+      if (pkg.target) {
+        pkg.name += '@' + pkg.target;
+      }
 
-  return task.run(taskOptions);
+      return pkg.name;
+    });
+  } else {
+    throw new SilentError('You must provide package to addAddonsToProject');
+  }
+
+  var install = (packages.length > 1) ? 'install addons' : 'install addon';
```

Comment:
Agreed, this is the existing pattern from addPackagesToProject. I can refactor all usages to something else.